x/crosschain/keeper/evm_hooks.go

```diff
@@ -78,47 +80,71 @@ func (k Keeper) ProcessLogs(
+	// read the logs and process inbounds from emitted events
+	// run the processing for the v1 and the v2 protocol contracts
 	for _, log := range logs {
-		eventZrc20Withdrawal, errZrc20 := ParseZRC20WithdrawalEvent(*log)
-		eventZetaSent, errZetaSent := ParseZetaSentEvent(*log, connectorZEVMAddr)
-		if errZrc20 != nil && errZetaSent != nil {
-			// This log does not contain any of the two events
-			continue
+		if !crypto.IsEmptyAddress(gatewayAddr) {
+			if err := k.ProcessZEVMInboundV2(ctx, log, gatewayAddr, emittingAddress, txOrigin); err != nil {
+				return errors.Wrap(err, "failed to process ZEVM inbound V2")
+			}
 		}
-		if eventZrc20Withdrawal != nil && eventZetaSent != nil {
-			// This log contains both events, this is not possible
-			ctx.Logger().
-				Error(fmt.Sprintf("ProcessLogs: log contains both ZRC20Withdrawal and ZetaSent events, %s , %s", log.Topics, log.Data))
-			continue
+		if err := k.ProcessZEVMInboundV1(ctx, log, connectorZEVMAddr, emittingAddress, txOrigin); err != nil {
+			return errors.Wrap(err, "failed to process ZEVM inbound V1")
 		}
+	}
 
-		// if eventZrc20Withdrawal is not nil we will try to validate it and see if it can be processed
-		if eventZrc20Withdrawal != nil {
-			// Check if the contract is a registered ZRC20 contract. If its not a registered ZRC20 contract, we can discard this event as it is not relevant
-			coin, foundCoin := k.fungibleKeeper.GetForeignCoins(ctx, eventZrc20Withdrawal.Raw.Address.Hex())
-			if !foundCoin {
-				ctx.Logger().
-					Info(fmt.Sprintf("cannot find foreign coin with contract address %s", eventZrc20Withdrawal.Raw.Address.Hex()))
-				continue
-			}
+	return nil
+}
 
-			// If Validation fails, we will not process the event and return and error. This condition means that the event was correct, and emitted from a registered ZRC20 contract
-			// But the information entered by the user is incorrect. In this case we can return an error and roll back the transaction
-			if err := k.ValidateZrc20WithdrawEvent(ctx, eventZrc20Withdrawal, coin.ForeignChainId); err != nil {
-				return err
-			}
-			// If the event is valid, we will process it and create a new CCTX
-			// If the process fails, we will return an error and roll back the transaction
-			if err := k.ProcessZRC20WithdrawalEvent(ctx, eventZrc20Withdrawal, emittingContract, txOrigin); err != nil {
-				return err
-			}
+// ProcessZEVMInboundV1 processes the logs emitted by the zEVM contract for V1 protocol contracts
+// it parses logs from Connector and ZRC20 contracts and processes them accordingly
+func (k Keeper) ProcessZEVMInboundV1(
+	ctx sdk.Context,
+	log *ethtypes.Log,
+	connectorZEVMAddr,
+	emittingAddress ethcommon.Address,
+	txOrigin string,
+) error {
+	eventZrc20Withdrawal, errZrc20 := ParseZRC20WithdrawalEvent(*log)
+	eventZetaSent, errZetaSent := ParseZetaSentEvent(*log, connectorZEVMAddr)
+	if errZrc20 != nil && errZetaSent != nil {
+		// This log does not contain any of the two events
+		return nil
+	}
```

Comment:
I copied existing logic, I agree it seems that it should be improved, but in practice nothing should happen because if error is non nil, the event will be nil, then the associated even will be skipped.

This code will be removed once we migrate the codebase to only parse v2 events